### PR TITLE
Add validation for 00173 & 00184

### PR DESCRIPTION
**Pull request by Tony-LunarG**
A portion of #6733

### CODE REVIEW

File: layers/core_checks/cc_copy_blit_resolve.cpp [hunk 1]
```diff
@@ -2424,6 +2432,49 @@ bool CoreChecks::ValidateImageSampleCount(const HandleT handle, const IMAGE_STAT
...
+        if (region.bufferRowLength != 0 && region.bufferImageHeight != 0) {
+            copy_size = ((region.bufferRowLength * region.bufferImageHeight) * element_size);
+        } else {
+            copy_size = ((region.imageExtent.width * region.imageExtent.height * region.imageExtent.depth) * element_size);
+        }
+        auto image_region = sparse_container::range<VkDeviceSize>{image_offset, image_offset + copy_size};
+        auto buffer_region = sparse_container::range<VkDeviceSize>{region.bufferOffset, region.bufferOffset + copy_size};
+        if (image_state.DoesResourceMemoryOverlap(image_region, &buffer_state, buffer_region)) {
+            const LogObjectList objlist(cb_state.commandBuffer(), image_state.image());
+            skip |= LogError(GetBufferMemoryImageCopyCommandVUID("00173", image_to_buffer, is_2), objlist, region_loc,
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_copy_blit_resolve.cpp:2471]: we should at least print which `pRegions` caused the error

Comment (Tony-LunarG) [reply layers/core_checks/cc_copy_blit_resolve.cpp:2471]: Isn't that what region_loc does?

Validation Error: [ VUID-vkCmdCopyImageToBuffer-pRegions-00184 ] Object 0: handle = 0x14df6e32900, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0xf56c9b0000000004, type = VK_OBJECT_TYPE_IMAGE; | MessageID = 0x45ef177c | vkCmdCopyImageToBuffer(): pRegions[0] Detected overlap between source and dest regions in memory. The Vulkan spec states: The union of all source regions, and the union of all destination regions, specified by the elements of pRegions, must not overlap in memory (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkCmdCopyImageToBuffer-pRegions-00184)

File: layers/core_checks/cc_copy_blit_resolve.cpp [hunk 2]
```diff
@@ -2424,6 +2432,49 @@ bool CoreChecks::ValidateImageSampleCount(const HandleT handle, const IMAGE_STAT
...
 
+template <typename RegionType>
+bool CoreChecks::ValidateImageBufferCopyMemoryOverlap(const CMD_BUFFER_STATE &cb_state, uint32_t regionCount,
+                                                      const RegionType *pRegions, const IMAGE_STATE &image_state,
+                                                      const BUFFER_STATE &buffer_state, const Location &loc, bool image_to_buffer,
+                                                      bool is_2) const {
+    bool skip = false;
+    auto element_size = vkuFormatElementSize(image_state.createInfo.format);
+    for (uint32_t i = 0; i < regionCount; ++i) {
+        const RegionType region = pRegions[i];
```
Comment (ziga-lunarg) [inline layers/core_checks/cc_copy_blit_resolve.cpp:2443]: Could this be a reference?

File: layers/core_checks/cc_copy_blit_resolve.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply layers/core_checks/cc_copy_blit_resolve.cpp:2471]: aww, I thought it was between 2 different regions,  lgtm

**Review by spencer-lunarg — APPROVED** [review]

**Pull request merged**